tools/test-js-with-node

```diff
@@ -290,12 +288,65 @@ def run_tests_via_node_js() -> int:
     ret = 0
     if parallel > 1:
         sub_tests = [test_files[i::parallel] for i in range(parallel)]
-        parallel_processes = [subprocess.Popen(node_tests_cmd + sub_test) for sub_test in sub_tests]
+        if options.coverage:
+            # ensure a clean start.
+            coverage_dir = os.path.join(ROOT_DIR, "var/node-coverage")
+            if os.path.exists(coverage_dir) and os.path.isdir(coverage_dir):
+                shutil.rmtree(coverage_dir)
+            temp_dir = os.path.join(ROOT_DIR, "var/node-coverage/nyc_outputs")
```

Comment:
I think we want to generate a random temporary directory like `var/node-coverage/nyc_outputs_abc123`; using a fixed directory means one can end up with data being accidentally shared between different runs of the program if you stopped it with `Ctrl+C`, for example.

Reply:
Well, we do use this to make sure that never happens:
```python
            # ensure a clean start.
            coverage_dir = os.path.join(ROOT_DIR, "var/node-coverage")
            if os.path.exists(coverage_dir) and os.path.isdir(coverage_dir):
                shutil.rmtree(coverage_dir)
```

but I guess stopping and continuing with `Cntrl-D` might be possible... I'll make a change to append `random.randrange(2**63)`, but also change the `join(ROOT_DIR, "var/node-coverage/nyc_outputs")` to be `join(coverage_dir , "nyc_outputs")`.